pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -571,31 +578,60 @@ impl InMemoryLayer {
         let mut before_segsizes;
         let mut after_page_versions;
         let mut after_segsizes;
+        let mut after_oldest_lsn;
         if !dropped {
             before_segsizes = BTreeMap::new();
             after_segsizes = BTreeMap::new();
+            after_oldest_lsn = None;
             for (lsn, size) in inner.segsizes.iter() {
-                if *lsn > end_lsn {
-                    after_segsizes.insert(*lsn, *size);
-                } else {
-                    before_segsizes.insert(*lsn, *size);
+                match lsn.cmp(&end_lsn) {
+                    Ordering::Less => {
+                        before_segsizes.insert(*lsn, *size);
+                    }
+                    Ordering::Equal => {
+                        // Page versions at the cutoff LSN will be stored in the
+                        // materialized image layer.
+                    }
```

Comment:
Don't these page versions need to exist in the temporary InMemoryLayers (at least before these temporary layers are replaced by historic on disk)

Reply:
Ah, quite right! I mixed up `freeze` and `write_to_disk`